client/asset/eth/rpcclient.go

```diff
@@ -8,213 +8,320 @@ package eth
 }
 
-// pendingTransactions returns pending transactions.
-func (c *rpcclient) pendingTransactions(ctx context.Context) ([]*types.Transaction, error) {
-	var ptxs []*types.Transaction
-	err := c.c.CallContext(ctx, &ptxs, "eth_pendingTransactions")
+// transactionReceipt uses a raw request to retrieve a transaction's receipt.
+func (n *nodeClient) transactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
+	_, blockHash, _, index, err := n.leth.ApiBackend.GetTransaction(ctx, txHash)
+	if err != nil {
+		return nil, nil
+	}
+	receipts, err := n.leth.ApiBackend.GetReceipts(ctx, blockHash)
 	if err != nil {
 		return nil, err
 	}
-	return ptxs, nil
+	if len(receipts) <= int(index) {
+		return nil, nil
+	}
+	return receipts[index], nil
+}
+
+// pendingTransactions returns pending transactions.
+func (n *nodeClient) pendingTransactions(ctx context.Context) ([]*types.Transaction, error) {
+	return n.leth.ApiBackend.GetPoolTransactions()
 }
 
 // addPeer adds a peer.
-func (c *rpcclient) addPeer(ctx context.Context, peer string) error {
-	return c.c.CallContext(ctx, nil, "admin_addPeer", peer)
+func (n *nodeClient) addPeer(ctx context.Context, peerURL string) error {
+	peer, err := enode.Parse(enode.ValidSchemes, peerURL)
+	if err != nil {
+		return err
+	}
+	n.p2pSrv.AddPeer(peer)
+	return nil
 }
 
 // blockNumber gets the block number at time of calling.
-func (c *rpcclient) blockNumber(ctx context.Context) (uint64, error) {
-	bn, err := c.ec.BlockNumber(ctx)
+func (n *nodeClient) blockNumber(ctx context.Context) (uint64, error) {
+	hdr, err := n.bestHeader(ctx)
 	if err != nil {
 		return 0, err
 	}
-	return bn, nil
+	return hdr.Number.Uint64(), nil
 }
 
 // nodeInfo retrieves useful information about a node.
-func (c *rpcclient) nodeInfo(ctx context.Context) (*p2p.NodeInfo, error) {
-	info := new(p2p.NodeInfo)
-	if err := c.c.CallContext(ctx, info, "admin_nodeInfo"); err != nil {
-		return nil, err
-	}
-	return info, nil
+func (n *nodeClient) nodeInfo(ctx context.Context) *p2p.NodeInfo {
+	return n.p2pSrv.NodeInfo()
 }
 
 // listWallets list all of the wallet's wallets? and accounts along with details
 // such as locked status.
-func (c *rpcclient) listWallets(ctx context.Context) ([]rawWallet, error) {
-	var res []rawWallet
-	if err := c.c.CallContext(ctx, &res, "personal_listWallets"); err != nil {
-		return nil, err
-	}
-	return res, nil
+func (n *nodeClient) listWallets(ctx context.Context) []accounts.Wallet {
+	return n.creds.ks.Wallets()
 }
 
 // sendTransaction uses a raw request to send tx.
-func (c *rpcclient) sendTransaction(ctx context.Context, tx map[string]string) (common.Hash, error) {
-	res := common.Hash{}
-	err := c.c.CallContext(ctx, &res, "eth_sendTransaction", tx)
+func (n *nodeClient) sendTransaction(ctx context.Context, txOpts *bind.TransactOpts,
+	to common.Address, data []byte) (*types.Transaction, error) {
+
+	nonce, err := n.leth.ApiBackend.GetPoolNonce(ctx, n.creds.addr)
```

Comment:
This appears to be what would have happened with the old method

https://github.com/ethereum/go-ethereum/blob/c10a0a62c3537fbcf899358e3c3d6be9507fa18c/internal/ethapi/transaction_args.go#L132-L138

Probably worth noting that internally, both `PrivateAccountAPI` and `PublicTransactionPoolAPI` do it the same way, but they use a mutex ('nonceLock') to sequence sign-send operations when a nonce is being generated with `setDefaults`. It might be worth adding some mutex protection to methods that generate transaction for the same reasons.